NDTensors/ext/NDTensorsCUDAExt/adapt.jl

```diff
@@ -3,12 +3,27 @@
 ## Will not allow us to properly utilize the buffer preference without changing the value of
 ## default_buffertype. Also `adapt(CuVector{<:Any, <:Any, Buffertype})` fails to work properly
 struct NDTensorCuArrayAdaptor{B} end
-## TODO make this work for unified. This works but overwrites CUDA's adapt_storage. This fails for emptystorage...
-function cu(xs; unified::Bool=false)
-  return fmap(
-    x -> adapt(NDTensorCuArrayAdaptor{unified ? Mem.UnifiedBuffer : Mem.DeviceBuffer}(), x),
-    xs,
+
+"""
+cu(xs; Buffer::String="Device")
+
+NDTensors version of `cu` function which preserves the number percision in the input.
+The array will use the buffer type Buffer which has option of `Device` = DeviceBuffer
+`Unified` = UnifiedBuffer and `Host` = HostBuffer
+"""
+function cu(xs; Buffer::String="Device")
+  buf = (
+    if Buffer == "Device"
+      Mem.DeviceBuffer
+    elseif Buffer == "Unified"
+      Mem.UnifiedBuffer
+    elseif Buffer == "Host"
+      Mem.HostBuffer
+    else
+      error("Buffer $(Buffer) is undefined")
+    end
```

Comment:
I don't really like moving to a string for this keyword argument, it's not very extensible (i.e. if `CUDA.jl` adds a new option we have to create a new string name and add it in by hand into this if-statement). Can we just keep it as-is for now and work on that change in a future PR?

My suggestion would be to just use `cu(xs; buffertype=Mem.DeviceBuffer)` and just expect that `buffertype` is a buffer type from `CUDA.jl`.

Reply:
(Or maybe `cu(xs; storagemode=Mem.DeviceBuffer)` if that's the name we are using for that elsewhere.)

Reply:
I changed this to use the `storagemode=Mem.DeviceBuffer` definition. Except I have an error where I named it `buffertype` I will fix when I migrate `storagemode` to `AbstractGPUArrays` tomorrow.